### PR TITLE
[MM-18963] Migrate components/local_date_time to TypeScript

### PR DESCRIPTION
**Pull request by kgeorgiou**
#### Summary

This migrates `components/local_date_time` component to TypeScript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12464

#### Related Pull Requests
N/A

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5496,6 +5505,7 @@
       "resolved": "https://registry.npmjs.org/bl/-/bl-1.2.2.tgz",
       "integrity": "sha512-e8tQYnZodmebYDWGH7KMRvtzKXaJHx3BbilrgZCfvyLUYdKpK1t5PSPmpkny/SgiTSCnjfLW7v5rlONXVFkQEA==",
       "dev": true,
+      "optional": true,
```
Comment (devinbinnie) [inline package-lock.json]: Can you remove all these little changes to `package-lock.json` and just keep the above change where the new package is added?

Comment (kgeorgiou) [reply package-lock.json]: Yep, sure thing!

`package-lock.json` may be worth a mention in the contributor guide 😅

File: package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
   "dependencies": {
     "@types/enzyme": "3.10.3",
     "@types/jest": "24.0.17",
+    "@types/moment-timezone": "^0.5.12",
```
Comment (crspeller) [inline package.json]: Use an exact version here please.

Comment (jasonblais) [reply package.json]: Hi @kgeorgiou, let us know if you have any questions? Would be happy to help! Thanks for your contribution :)

Comment (kgeorgiou) [reply package.json]: Oops sorry, I missed this comment! Will take care of it later today :)

**Pull request merged**